http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -662,6 +682,88 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+    ev: Ask[F, AppContext]
+  ): F[LandingZoneResources] =
+    for {
+      // Step 1: call LZ for LZ id
+      landingZoneOpt <- wsmDao.getLandingZone(billingProfileId, userToken)
+      landingZone <- F.fromOption(
+        landingZoneOpt,
+        AppCreationException(s"Landing zone not found for billing profile ${billingProfileId}")
+      )
+      landingZoneId = landingZone.landingZoneId
+
+      // Step 2: call LZ for LZ resources
+      lzResourcesByPurpose <- wsmDao.listLandingZoneResourcesByType(landingZoneId, userToken)
+
+      aksClusterName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                   "Microsoft.ContainerService/managedClusters",
+                                                   SHARED_RESOURCE.toString
+      )
+      batchAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                     "Microsoft.Batch/batchAccounts",
+                                                     SHARED_RESOURCE.toString
+      )
+      relayNamespace <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                   "Microsoft.Relay/namespaces",
+                                                   SHARED_RESOURCE.toString
+      )
+      storageAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                       "Microsoft.Storage/storageAccounts",
+                                                       SHARED_RESOURCE.toString
+      )
+      vnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                             "Microsoft.Network/virtualNetworks",
+                                             SHARED_RESOURCE.toString
+      )
+      batchNodesSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                         "Microsoft.Network/virtualNetworks/subnets",
+                                                         WORKSPACE_BATCH_SUBNET.toString
+      )
+      aksSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                  "Microsoft.Network/virtualNetworks/subnets",
+                                                  AKS_NODE_POOL_SUBNET.toString
+      )
+    } yield LandingZoneResources(
+      AKSClusterName(aksClusterName),
+      BatchAccountName(batchAccountName),
+      RelayNamespace(relayNamespace),
+      StorageAccountName(storageAccountName),
+      NetworkName(vnetName),
+      SubnetworkName(batchNodesSubnetName),
+      SubnetworkName(aksSubnetName)
+    )
+
+  private def getLandingZoneResourceName(landingZoneResourcesByPurpose: List[LandingZoneResourcesByPurpose],
+                                         resourceType: String,
+                                         purpose: String
+  ): F[String] = {
+    val lzResourcesOpt =
```

Comment:
What about this one-liner :) 
```
    landingZoneResourcesByPurpose
      .filter(_.purpose == purpose)
      .flatMap(_.deployedResources)
      .filter(_.resourceType.equalsIgnoreCase(resourceType))
      .headOption
      .flatMap(r => r.resourceName.orElse(r.resourceId.flatMap(_.split('/').lastOption)))
      .fold(
        F.raiseError[String](
          AppCreationException(s"${resourceType} resource with purpose ${purpose} not found in landing zone")
        )
      )(F.pure)
```

To me it reads a little cleaner, but don't feel too strongly (I think the logic is the same).